compiler/rustc_graphviz/src/lib.rs

```diff
@@ -37,6 +37,10 @@
 //! ```rust
 //! #![feature(rustc_private)]
 //!
+//! #[allow(unused_extern_crates)]
+//! // extern `rustc_graphviz` explicitly to avoid lint error
+//! extern crate rustc_graphviz;
```

Comment:
Can you remove it alltogether? The comment and your `allow` seem to disagree with each other.

Reply:
I had to implement that lines to avoid following error:

```text
failures:

---- src/lib.rs - (line 37) stdout ----
error: `extern crate` is not idiomatic in the new edition
 --> src/lib.rs:39:1
  |
5 | extern crate r#rustc_graphviz;
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: convert it to a `use`
  |
note: the lint level is defined here
 --> src/lib.rs:36:9
  |
2 | #![deny(warnings)]
  |         ^^^^^^^^
  = note: `#[deny(unused_extern_crates)]` implied by `#[deny(warnings)]`

error: aborting due to previous error

Couldn't compile the test.

failures:
    src/lib.rs - (line 37)
```

Even the allow itself doesn't help, following extern implementation has to be there for fix. That's the workaround I found, not sure if there is a better way.

Reply:
oh ugh, the `r#rustc_graphviz` is injected by rustdoc itself, I don't think there's a way to avoid the lint triggering. Maybe rustdoc could inject the allow for just the `extern crate` statement so it doesn't affect any other part of the code?

Reply:
> oh ugh, the `r#rustc_graphviz` is injected by rustdoc itself, I don't think there's a way to avoid the lint triggering. Maybe rustdoc could inject the allow for just the `extern crate` statement so it doesn't affect any other part of the code?

Hmm, sounds like better solution.

```diff
diff --git a/src/librustdoc/doctest.rs b/src/librustdoc/doctest.rs
index 81d9c46447a..7f9ac790d1d 100644
--- a/src/librustdoc/doctest.rs
+++ b/src/librustdoc/doctest.rs
@@ -682,6 +682,8 @@ pub(crate) fn make_test(
             // parse the source, but only has false positives, not false
             // negatives.
             if s.contains(crate_name) {
+                // Allow `unused_extern_crates` lint rule to avoid lint errors/warnings
+                prog.push_str(&format!("#[allow(unused_extern_crates)]\n"));
                 prog.push_str(&format!("extern crate r#{crate_name};\n"));
                 line_offset += 1;
             }
```

This handles the issue